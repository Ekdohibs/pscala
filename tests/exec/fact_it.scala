class Fact {
  def fact(n: Int) : Int = {
    var x = n;
    var res = 1;
    while (x > 1) {
      res = res * x;
      x = x-1
    };
    res
  }
}

object Main {
  def main(args: Array[String]) {
    var f = new Fact();
    print(f.fact(5)); print("\n");
    print(f.fact(11)); print("\n")
  }
}
