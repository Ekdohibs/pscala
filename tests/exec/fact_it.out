120
39916800
